src/UNL/Peoplefinder/Record/Avatar.php

```diff
@@ -155,17 +161,26 @@ protected function generatePersonUrl($options)
         $planetRedUid = $this->record->getProfileUid();
         $profileIconUrl = UNL_Peoplefinder_Record::PLANETRED_BASE_URL . 'icon/' . 'unl_' . $planetRedUid . '/' . $size . '/';
 
-        $request = new HTTP_Request2($profileIconUrl, HTTP_Request2::METHOD_HEAD, [
-            'adapter' => 'HTTP_Request2_Adapter_Curl',
-            'follow_redirects' => true,
-            'strict_redirects' => true,
+        $effectiveUrl = $profileIconUrl;
+        $onRedirect = function(
+            RequestInterface $request,
+            ResponseInterface $response,
+            UriInterface $uri
+        ) use (&$effectiveUrl) {
+            $effectiveUrl = (string) $uri;
```

Comment:
Unfortunately, the response object does not have that API. The only way to get the URI is by examining the request. The client frequently modifies the initial request (by cloning) leaving no way to get the _final_ version of the request that was used with the response. Guzzle just doesn't want to make it easy ;)